packages/types/src/lib/interfaces.ts

```diff
@@ -1545,3 +1531,84 @@ export interface MintCapacityCreditsRes {
   capacityTokenId: any;
   capacityTokenIdStr: string;
 }
+
+/**
+ * ========== Siwe Messages ==========
+ */
+export interface BaseSiweMessage {
+  walletAddress: string;
+  nonce: string;
+
+  // -- filled in by default
+  expiration?: string;
+  resources?: LitResourceAbilityRequest[];
+  uri?: string; // This is important in authNeededCallback params eg. (lit:session:xxx)
+  domain?: string;
+  statement?: string;
+  version?: string;
+  chainId?: number;
+  litNodeClient?: any;
+}
+
+export interface WithRecap extends BaseSiweMessage {
+  uri: string;
+  expiration: string;
+  resources: LitResourceAbilityRequest[];
+}
```

Comment:
Why do we need to define this? Can't we use it from the siwe library?

Reply:
because we are running custom type `LitResourceAbilityRequest`